src/ape/api/transactions.py

```diff
@@ -243,3 +274,219 @@ def await_confirmations(self) -> "ReceiptAPI":
+        contract_type = self._receipt.chain_manager.contracts.get(address)
+        call_signature = None
+        if contract_type:
+            selector = call.calldata[:4].hex()
+            method = _get_method_called(selector, contract_type)
+            if method:
+                arguments = self._decode_calldata(method, call.calldata[4:])
+                return_value = self._decode_returndata(method, call.returndata)
+                call_signature = str(
+                    _MethodTraceSignature(contract_type.name, method.name, arguments, return_value)
+                )
+
+        call_signature = call_signature or next(call.display_nodes).title  # type: ignore
+
+        if call.failed:
+            call_signature = f"[{_FAILURE_TRACE_COLOR}]{call_signature}"
+
+        parent = Tree(call_signature, guide_style="dim")
+        for sub_call in call.calls:
+            parent.add(self.create_tree(sub_call))
+
+        return parent
+
+    def _decode_calldata(self, method: MethodABI, raw_data: bytes) -> Dict:
+        input_types = [i.canonical_type for i in method.inputs]  # type: ignore
+
+        try:
+            raw_input_values = decode_abi(input_types, raw_data)
+            input_values = [
+                self._decode_value(
+                    self._ecosystem.decode_primitive_value(v, parse_type(t)),
+                )
+                for v, t in zip(raw_input_values, input_types)
+            ]
+        except InsufficientDataBytes:
+            input_values = ["<?>" for _ in input_types]
+
+        arguments = {}
+        index = 0
+        for i, v in zip(method.inputs, input_values):
+            name = i.name or f"{index}"
+            arguments[name] = v
+            index += 1
+
+        return arguments
+
+    def _decode_returndata(self, method: MethodABI, raw_data: bytes) -> Any:
+        values = [
+            self._decode_value(v) for v in self._ecosystem.decode_returndata(method, raw_data)
+        ]
+
+        if len(values) == 1:
+            return values[0]
+
+        return values
+
+    def _decode_value(self, value):
+        decoded_value = value
+        if isinstance(value, HexBytes):
+            try:
+                string_value = value.strip(b"\x00").decode("utf8")
+                decoded_value = f"'{string_value}'"
+            except UnicodeDecodeError:
+                decoded_value = humanize_hash(value)
+
+        elif isinstance(value, str) and value.startswith("0x"):
+            decoded_value = value
+
+        elif isinstance(value, str):
+            # Surround non-address strings with quotes.
+            decoded_value = f'"{value}"'
+
+        elif isinstance(value, (list, tuple)):
+            return [self._decode_value(v) for v in value]
+
+        elif isinstance(value, Struct):
+            return {k: v for k, v in value.items()}
+
+        return decoded_value
+
+
+@dataclass()
+class _MethodTraceSignature:
```

Comment:
This might be a good util method?

Reply:
Perhaps! I tend to prefer moving things to `utils` on an as-needs basis. In the future, if something needs this functionality, we may need to adjust it slightly before making it public.

Reply:
ping on this